src/sidebar/util/term-before-position.ts

```diff
@@ -0,0 +1,12 @@
+/**
+ * Given a text, this returns the "word" right before a specific position.
+ *
+ * In this context, a word is anything between a space or newline, and provided
+ * position.
+ */
+export function termBeforePosition(text: string, position: number): string {
+  const textBeforePosition = text.slice(0, position).replace(/\n/g, ' ');
+  const lastSpaceIndex = textBeforePosition.lastIndexOf(' ');
+
+  return textBeforePosition.slice(lastSpaceIndex + 1);
```

Comment:
> This might be useful if the set of "space" characters needs to be expanded.

Which we probably should, if we want to treat tabs and similar characters the same.

Reply:
I had to get that regular expression explained by ChatGPT 😅. But it does make sense, and all tests keep passing, so I have changed it